src/spicedb-common/parsers/dsl/dsl.ts

```diff
@@ -105,26 +105,30 @@ export function flatMapExpression<T>(
     case 'namedarrow':
       // fallthrough
       
-    case 'arrow':
+    case 'arrow': {
```

Comment:
Adding brackets provides proper block scoping for the variable declarations within.